### DISCUSSION TITLE
1.4.0 upgrade, unable to upgrade Engine

### DISCUSSION THREAD

**Discussion by xamindar** [longhorn/longhorn#5207] [General]
I just performed the 1.4.0 Manager upgrade. When following the next section for upgrading the Engine of each volume, I have nothing to select for engine.

Clicking the Volume -> three line button -> Upgrade Engine
The resulting popup has an image list drop-down that is empty. So I am unable to select anything to upgrade to.

If I go over to Settings in the longhorn UI and select "Engine Image", I do see my two images listed there. My old one and a new "longhornio/longhorn-engine:v1.4.0" image which is set to default. 

Did I miss a step somewhere?

Also, I have verified all the pods are running (or completed) regarding longhorn using "kubectl get pods -n longhorn-system". So nothing is stuck in error afaik.

**Comment by innobead** [comment]
@xamindar Please provide the support bundle for the further check. Not sure if this is caused by the engine image is not ready. 

cc @c3y1huang @longhorn/qa

**Comment by xamindar** [comment]
I came back 3 hours later and checked the longhorn gui and I no longer see the old engine listed in the "engine image" setting screen. Also, the upgrade engine option is grayed out on all volumes.
Checking my volumes they show they are on "longhornio/longhorn-engine:v1.4.0".

So did it just upgrade automatically in the background? I'm a little confused as this was not mentioned anywhere.

In any case, if you still wanted to take a look, here is the support bundle:
[supportbundle_2724eb30-abc5-4e93-8bde-5684aa6238ae_2023-01-05T01-26-46Z.zip](https://github.com/longhorn/longhorn/files/10348658/supportbundle_2724eb30-abc5-4e93-8bde-5684aa6238ae_2023-01-05T01-26-46Z.zip)

**Comment by c3y1huang** [comment]
> So did it just upgrade automatically in the background? I'm a little confused as this was not mentioned anywhere.

Base on your support-bundle, the `concurrent-automatic-engine-upgrade-per-node-limit` is set to 1, this means Longhorn will automatically update volumes engines.
```yaml
- apiVersion: longhorn.io/v1beta2
  kind: Setting
  metadata:
    annotations:
      longhorn.io/configmap-resource-version: "27706"
    creationTimestamp: "2022-08-10T01:27:45Z"
    generation: 2
    managedFields:
    - apiVersion: longhorn.io/v1beta2
      fieldsType: FieldsV1
      fieldsV1:
        f:metadata:
          f:annotations:
            .: {}
            f:longhorn.io/configmap-resource-version: {}
        f:value: {}
      manager: longhorn-manager
      operation: Update
      time: "2022-08-10T01:27:45Z"
    name: concurrent-automatic-engine-upgrade-per-node-limit
    namespace: longhorn-system
    resourceVersion: "33735817"
    uid: 84fc3a60-11f6-4552-b8ac-1eac1abcb0a9
  value: "1"
```
You can set to 0 to disable auto-upgrade. Ref: [Concurrent Automatic Engine Upgrade Per Node Limit](https://longhorn.io/docs/1.4.0/references/settings/#concurrent-automatic-engine-upgrade-per-node-limit)

**Comment by c3y1huang** [comment]
The auto engine upgrade is expected behavior by setting. Closing issue.
